src/lib/components/ServerAddressModal.svelte

```diff
@@ -31,29 +33,53 @@
   }
 
   function clearSettings() {
+    tabAsSpaces = true;
     serverAddress = null;
     headers = null;
-    serverSettings.set({address: null, status: 'disconnected', options:{}, id:undefined});
+    serverSettings.set({
+      address: null,
+      status: "disconnected",
+      options: {},
+      id: undefined,
+    });
     localStorage.removeItem("address");
     settingsModal = false;
   }
 
-  function handleTextArea(e){
-    if(e.code == 'Tab') {e.preventDefault() 
-      headers = headers + '    ';
-    };
-  }
+  function handleTextArea(e) {
+    if (e.code == "Tab" && tabAsSpaces) {
+      e.preventDefault();
+      headers = headers + "    ";
+    } else if (e.code == "KeyM" && e.ctrlKey) {
+      e.preventDefault();
+      tabAsSpaces = !tabAsSpaces;
 
+      statusMessage = tabAsSpaces
+        ? "Pressing Tab will now insert the tab character."
+        : "Pressing Tab will now move focus to the next focusable element.";
+        setTimeout(() => {
+          statusMessage = '';
+        }, 200);
+    }
+  }
   onMount(() => {
     serverAddress = localStorage.getItem("address");
   });
 </script>
 
-<button class="pr-2 w-full text-center text-clip overflow-hidden" on:click={() => (settingsModal = !settingsModal)}>
-    { serverAddress ||'URL'}
+<div role="status" aria-live="polite" class="sr-only">{statusMessage}</div>
+<button
+  class="pr-2 w-full text-center text-clip overflow-hidden"
+  on:click={() => (settingsModal = !settingsModal)}
+>
+  {serverAddress ? "Current URL: " + serverAddress : "Set URL"}
```

Comment:
"Set URL" is a good idea, but I think there's not enough space for "Current URL:" in this part .

![](https://i.ibb.co/R9DWXdj/Screenshot-20230812-064540.png)

```suggestion
  {serverAddress ?  serverAddress : "Set URL"}
```

Reply:
Does this work better? it should be like it was before but also for screen readers it would be a more descriptive link.
```

<button
  class="pr-2 w-full text-center text-clip overflow-hidden"
  on:click={() => (settingsModal = !settingsModal)}
>
  <span class="sr-only">{serverAddress ? "Current URL: " + serverAddress : "Set URL"}</span>
  <span aria-hidden="true">{serverAddress || "Set URL"}</span>
</button>
```

Reply:
You're right. What about "URL" instead of "Current URL"?
BTW, I also have to make it single line.

Reply:
Sorry I am blind, the button with the spans makes it not same as it wass before? I thought sr-only makes the content not visible for sighted people, an aria-hidden makes it not visible for blind, so we can have both solutions together.